app/src/main/java/com/nutomic/syncthingandroid/activities/DeviceActivity.java

```diff
@@ -293,18 +299,26 @@ private void onApiChange(SyncthingService.State currentState) {
     }
 
     private void updateViewsAndSetListeners() {
+        mIdView.removeTextChangedListener(mIdTextWatcher);
+        mNameView.removeTextChangedListener(mNameTextWatcher);
+        mAddressesView.removeTextChangedListener(mAddressesTextWatcher);
+        mIntroducerView.setOnCheckedChangeListener(null);
+        mDevicePaused.setOnCheckedChangeListener(null);
```

Comment:
Why are we touching mIdView and others here?

Reply:
Removing those listeners temporarily prevents the "user changed property" function to trigger if we fill or update the UI. Thats why the function is called updateViewsAndSetListeners. Look a litte bit down, they are readded after the fields have been filled.

Reply:
Right, but what in this PR prompted the need for this change?

Reply:
The new 'consolidated' checkedlistener does not compare previous and new values to make the code simpler. This does not work if the no change event will trigger from the above ui init function. I just did it the same way as it was aleady in FolderActivity before this pr.